hypervisor/src/kvm/mod.rs

```diff
@@ -596,15 +595,24 @@ impl cpu::Vcpu for KvmVcpu {
     /// let vcpu = vm.create_vcpu(0).unwrap();
     /// let state = vcpu.cpu_state().unwrap();
     ///
-    fn cpu_state(&self) -> cpu::Result<CpuState> {
+    fn cpu_state(&self, msr_list: &MsrList) -> cpu::Result<CpuState> {
```

Comment:
I think it'd be cleaner if we could have a:

```rust
 #[cfg(target_arch = "x86_64")]
 msrs: MsrEntries,
```

in the `KvmVcpu` structure, that would cache the `MsrEntries` array creation that we do below. The list of MSRs from KVM is only created once at boot/restore so we don't need to generate the `MsrEntries` array each time we get the CPU state.
Moreover, having it as part of the `KvmCpu` structure would keep the `cpu_state` method from the `Vcpu` trait architecture agnostic.

Reply:
I can do what you're describing, but let me explain why I chose to go this way.
The list of MSRs that we want to save should be specific to the caller (`vmm` crate in this case). By implementing this as part of the `KvmCpu` would simplify the code but it would force anyone using the `hypervisor` crate to save all MSRs, which might not be what other VMM expect.
WDYT?

Reply:
I understand, but is there a use case, with KVM, for not saving/restoring all the MSRs KVM supports? We've seen the side effects of trying to only use a reduced set of MSRs ;-)

If at some point we think the list of MSRs we save/restore should be VM/VMM specific, then we could extend the `Vm` trait with a `set_msr_entries(MsrEntries)` method and use the entire KVM list of MSRs by default. But until we have a use case for that, I'd suggest to keep it simple and just save/restore all MSRs.

Reply:
Sounds good :+1: